python/cudf/cudf/core/column/column.py

```diff
@@ -1948,9 +1948,17 @@ def as_column(
                 raise TypeError(
                     f"Cannot convert a {inferred_dtype} of object type"
                 )
+            elif (
+                nan_as_null is False
+                and inferred_dtype in {"string"}
+                and pd.Series(arbitrary)
+                .apply(lambda x: isinstance(x, float) and np.isnan(x))
+                .any()
```

Comment:
I played with this a bit on my side, before you opened this PR -- I feel like there must be a faster way. This takes 2.06 seconds to run on the sample dataset we were looking at. That's a majority of the time spent in the operation, which takes 2.86 seconds to do the entire process (copy CPU to GPU and execute the GPU code).

This snippet took 1.03 seconds (twice as fast, and it can early-exit unlike the current code). Can we do even better? It's a very costly check, so optimization is worthwhile here. We should also evaluate if there are other checks we can evaluate first, and defer this as much as possible.
```suggestion
                and any(isinstance(x, float) and np.isnan(x) for x in arbitrary)
```

Reply:
yea :/ I don't like the solution due to it's inefficiency but I couldn't come up with any public API to achieve this check. @mroeschke Do you have a better solution for this check?